### PR TITLE
(DOCSP-21382): New section with embedded object relationship examples

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-21382
- https://jira.mongodb.org/browse/DOCSP-21383

### Staged Changes (Requires MongoDB Corp SSO)

- [Relationships](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-21382/schemas/relationships/#embedded-object-relationships-with-foreign-collections): New section for Embedded Object Relationships

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/schemas/relationships.txt [hunk 1]
```diff
@@ -238,6 +238,91 @@ or a null value in resolved GraphQL types and SDK models:
            }
          }
 
+Embedded Object Relationships
+-----------------------------
+
+Embedded Objects can have relationships with foreign collections. Use 
+dot notation to access properties in embedded objects when creating
+relationships with foreign collections.
```
Comment (nathan-contino-mongo) [inline source/schemas/relationships.txt:246]:
```suggestion
dot notation to access properties in embedded objects.
```

Maybe I'm missing something, but the extra clause on this sentence:
- doesn't seem to add anything (don't you *always* use dot notation when referencing embedded objects?)
- makes the sentence less readable

So I suggest removing it. If it does add something, maybe we can break the sentence up?

File: source/schemas/relationships.txt [hunk 2]
```diff
@@ -238,6 +238,91 @@ or a null value in resolved GraphQL types and SDK models:
...
+
+.. _relationships-from-embedded-object-to-another-collection:
+
+Embedded Object to Another Collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An embedded object can have a relationship with an object in a foreign 
+collection.
```
Comment (nathan-contino-mongo) [inline source/schemas/relationships.txt:254]:
```suggestion
collection:
```

Maybe other examples on this page don't use colons for the transition, but it feels very natural here. Could always update the other examples, too.

File: source/schemas/relationships.txt [hunk 3]
```diff
@@ -238,6 +238,91 @@ or a null value in resolved GraphQL types and SDK models:
...
+             "favoriteToyBrand": { "bsonType": "objectId" }
+           }
+         }
+       }
+      // ...additional model properties
+     }
+   }
+
+The dot notation to access a embedded object property contained in a list 
+is: ``pets.[].favoriteToyBrand``. From there, you can specify the
```
Comment (nathan-contino-mongo) [inline source/schemas/relationships.txt:313]:
```suggestion
To access an embedded object property contained in a list, use ``.[].``, e.g.
``pets.[].favoriteToyBrand``. From there, you can specify the 
```
Passive -> active reword suggestion. Take it or leave it or change it, up to you.
"a" -> "an", should probably take it.

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM with some smol nits. Nice work!

File: source/schemas/relationships.txt [hunk 4]
```diff
@@ -238,6 +238,90 @@ or a null value in resolved GraphQL types and SDK models:
...
+         "foreign_key": "_id", 
+         "is_list": false }
+   }
+
+.. _relationships-to-embedded-object-within-list:
+
+Embedded Object Within a List
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An embedded object that is within a list property can have a relationship
```
Comment (tkaye407) [inline source/schemas/relationships.txt:289]: Looks good to me. Dont think you need another example but you can add a note that dictionaries and sets use the same [] syntax

**Review by tkaye407 — APPROVED** [review]

File: source/schemas/relationships.txt [hunk 5]
```diff
@@ -238,6 +238,90 @@ or a null value in resolved GraphQL types and SDK models:
...
+``field1.[].field2``, e.g. ``pets.[].favoriteToyBrand``. From there, 
+you can specify the foreign collection details and foreign key field.
+
+.. code-block:: json
+
+   {
+     "pets.[].favoriteToyBrand": {
+       "ref": "#/relationship/mongodb-atlas/example/ToyBrand",
+       "foreign_key": "_id",
+       "is_list": false
```
Comment (kmorkos) [inline source/schemas/relationships.txt:321]: @tkaye407 is this supposed to be `"is_list": true`?

File: source/schemas/relationships.txt (hunk 4 above)
Comment (dacharyc) [reply source/schemas/relationships.txt:289]: Good note, I'll add that - thanks!

File: source/schemas/relationships.txt (hunk 5 above)
Comment (dacharyc) [reply source/schemas/relationships.txt:321]: I had assumed in this example we're referring to the foreign_key field as a list (or not) - in which case the ToyBrand `_id` is unlikely to be a list. But it's very possible my understanding is backwards!

Comment (tkaye407) [reply source/schemas/relationships.txt:321]: Nope. Is_List refers to if the field at the end of the relationship string is a list. So if favoriteToyBranc was itself a list / set / dictionary, then is_list would need to be true. Here, the property being pointed to: "pets.[].favoriteToyBrand" is just a primitive

Comment (tkaye407) [reply source/schemas/relationships.txt:321]: Perhaps its worth clarifying (a) that is_list is also true for sets / dictionaries and (b) that it refers to whether the property itself is a list / set / dictionary, not if it is withing a list of embedded objects

Comment (kmorkos) [reply source/schemas/relationships.txt:326]: ah i see the distinction - thanks for confirming!

**Pull request merged**